spinta/dimensions/param/ufuncs.py

```diff
@@ -1,10 +1,140 @@
-from typing import Any
+from collections.abc import Generator
+from typing import Any, Iterator
 
-from spinta.core.ufuncs import Bind
+from spinta import commands
+from spinta.components import Model
+from spinta.core.ufuncs import Bind, Expr
 from spinta.core.ufuncs import Env
 from spinta.core.ufuncs import ufunc
+from spinta.datasets.components import Param
+from spinta.dimensions.param.components import ParamBuilder, ParamLoader
+from spinta.utils.schema import NotAvailable
+
+
+class _Loop:
+    expr: Expr
+    arg: Any
+
+    def __init__(self, expr, arg):
+        self.arg = arg
+        self.expr = expr
 
 
 @ufunc.resolver(Env, Bind)
 def param(env: Env, bind: Bind) -> Any:
     return env.params[bind.name]
+
+
+@ufunc.resolver(ParamBuilder, Expr)
+def select(env: ParamBuilder, expr: Expr) -> Any:
+    if isinstance(env.this, Model):
+        data = commands.getall(env.context, env.this, env.this.backend, query=expr, resolved_params=env.params)
+        return data
+
+
+@ufunc.resolver(ParamBuilder, Expr, name="getattr")
+def getattr_(env: ParamBuilder, expr: Expr):
+    args, kwargs = expr.resolve(env)
+    resolved = env.call("getattr", *args)
+    yield from resolved
+
+
+@ufunc.resolver(ParamBuilder, Generator, object, name="getattr")
+def getattr_(env: ParamBuilder, generator: Generator, obj: object):
+    for item in generator:
+        yield from env.call("getattr", item, obj)
+
+
+@ufunc.resolver(ParamBuilder, _Loop, Bind, name="getattr")
+def getattr_(env: ParamBuilder, _loop: _Loop, bind: Bind):
+    data = env.call('getattr', _loop.arg, bind)
+    if isinstance(data, Iterator):
+        for item in data:
+            if item:
+                yield item
+
+                if item != env.params[env.target_param]:
+                    env.params[env.target_param] = item
+                    args, kwargs = _loop.expr.resolve(env)
+                    for arg in args:
+                        yield from env.call('getattr', _Loop(_loop.expr, arg), bind)
+
+
+@ufunc.resolver(ParamBuilder, dict, Bind, name="getattr")
+def getattr_(env: ParamBuilder, data: dict, bind: Bind):
+    if bind.name in data:
+        yield data[bind.name]
+
+
+@ufunc.resolver(ParamBuilder, Expr)
+def loop(env: ParamBuilder, expr: Expr):
+    args, kwargs = expr.resolve(env)
+    for arg in args:
+        yield _Loop(expr, arg)
+
+
+@ufunc.resolver(ParamBuilder, Generator, Bind)
+def loop(env: ParamBuilder, generator: Generator, bind: Bind):
+    for item in generator:
+        data = env.call("getattr", item, bind)
+        yield data
```

Comment:
There should be no such function like this. Here you are trying to resolve `loop(rows, attr)`. But instead it should be `getattr(loop(rows), attr)`.

`getattr` should expect generator type, iterate over it and call getattr again with a single row, for example:

```python
def getattr(env, rows, bind):
    for row in rows:
        env.call('getattr', row, bind)
```

Reply:
loop is no longer part of the implementation